lib/components/base-components/NormalComponent/NormalComponent_doInitialSilkscreenOverlapAdjustment.ts

```diff
@@ -0,0 +1,342 @@
+          anchor_position: flippedPosition,
+        })
+      }
+    }
+  }
+}
+
+/**
+ * Get all silkscreen elements from database records
+ * Uses selectAll for component discovery but database records for accurate bounds
+ * This is a compromise approach until component bounds are fully implemented
+ */
+function getAllSilkscreenElementsFromDB(
+  rootCircuit: any,
+  db: any,
+): SilkscreenElementWithBounds[] {
+  const elements: SilkscreenElementWithBounds[] = []
+
+  // Use selectAll to discover which components exist, then get their DB records
+  const silkscreenTexts = rootCircuit.selectAll("silkscreentext")
+  const silkscreenPaths = rootCircuit.selectAll("silkscreenpath")
+  const silkscreenRects = rootCircuit.selectAll("silkscreenrect")
+  const silkscreenCircles = rootCircuit.selectAll("silkscreencircle")
+
+  // Process silkscreen texts
+  for (const textComp of silkscreenTexts) {
+    const dbRecord = textComp.pcb_silkscreen_text_id
+      ? db.pcb_silkscreen_text.get(textComp.pcb_silkscreen_text_id)
+      : null
+    if (dbRecord) {
+      elements.push({
+        bounds: getSilkscreenTextBounds(dbRecord),
+        layer: dbRecord.layer,
+        type: "text",
+        pcb_component_id: dbRecord.pcb_component_id,
+        element_id: dbRecord.pcb_silkscreen_text_id,
+      })
+    }
+  }
+
+  // Process silkscreen paths
+  for (const pathComp of silkscreenPaths) {
+    const dbRecord = pathComp.pcb_silkscreen_path_id
+      ? db.pcb_silkscreen_path.get(pathComp.pcb_silkscreen_path_id)
+      : null
+    if (dbRecord) {
+      elements.push({
+        bounds: getSilkscreenPathBounds(dbRecord),
+        layer: dbRecord.layer,
+        type: "path",
+        pcb_component_id: dbRecord.pcb_component_id,
+        element_id: dbRecord.pcb_silkscreen_path_id,
+      })
+    }
+  }
+
+  // Process silkscreen rects
+  for (const rectComp of silkscreenRects) {
+    const dbRecord = rectComp.pcb_silkscreen_rect_id
+      ? db.pcb_silkscreen_rect.get(rectComp.pcb_silkscreen_rect_id)
+      : null
+    if (dbRecord) {
+      elements.push({
+        bounds: getSilkscreenRectBounds(dbRecord),
+        layer: dbRecord.layer,
+        type: "rect",
+        pcb_component_id: dbRecord.pcb_component_id,
+        element_id: dbRecord.pcb_silkscreen_rect_id,
+      })
+    }
+  }
+
+  // Process silkscreen circles
+  for (const circleComp of silkscreenCircles) {
+    const dbRecord = circleComp.pcb_silkscreen_circle_id
+      ? db.pcb_silkscreen_circle.get(circleComp.pcb_silkscreen_circle_id)
+      : null
+    if (dbRecord) {
+      elements.push({
+        bounds: getSilkscreenCircleBounds(dbRecord),
+        layer: dbRecord.layer,
+        type: "circle",
+        pcb_component_id: dbRecord.pcb_component_id,
+        element_id: dbRecord.pcb_silkscreen_circle_id,
+      })
+    }
+  }
+
+  return elements
+}
+
+/**
+ * Get component center using component's built-in bounds system
+ */
+function getComponentCenterFromCache(
```

Comment:
THERES NO CACHE! Don't lie in your function names! Also don't use a cache!!! Also there's no need for this function!!!! Remove it, it also shouldn't take the unused db!!!! `component._getPcbCircuitJsonBounds().center` is extremely clear